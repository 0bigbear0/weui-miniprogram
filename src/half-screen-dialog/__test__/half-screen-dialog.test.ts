import path from 'path'
import simulate from 'miniprogram-simulate'

describe('grids', () => {
    const id = simulate.load({
        compiler: 'official',
        rootPath: __dirname,
        template: `
            <mp-half-screen-dialog show="{{show}}">
                <view slot="title">测试标题A</view>
                <view slot="desc">这是自定义描述区</view>
                <view slot="footer">
                    <button type="default" class="weui-btn">按钮A</button>
                    <button type="primary" class="weui-btn">按钮B</button>
                </view>
            </mp-half-screen-dialog>
        `,
        usingComponents: {
            'mp-half-screen-dialog': simulate.load(path.resolve(__dirname, '../half-screen-dialog'))
        },
        data: {
            show: false
        }
    })

    test('basic', async () => {
        const comp = simulate.render(id)
        comp.attach(document.createElement('parent-wrapper'))
        await simulate.sleep(0)
        expect(comp.toJSON()).toMatchSnapshot()

        comp.setData({ show: true })
        await simulate.sleep(0)
        expect(comp.toJSON()).toMatchSnapshot()
    })
})
