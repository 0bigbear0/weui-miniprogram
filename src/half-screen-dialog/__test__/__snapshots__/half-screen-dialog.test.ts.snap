// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`grids basic 1`] = `
<main>
  <wx-view
    class="weui-hidden"
  >
    <wx-view
      class="weui-mask init"
      data-type="tap"
      bind:tap="close"
    />
    <wx-view
      class="weui-half-screen-dialog "
    >
      <wx-view
        class="weui-half-screen-dialog__hd"
      >
        <wx-view
          class="weui-half-screen-dialog__hd__side"
          data-type="close"
          bind:tap="close"
        >
          <wx-view
            class="weui-icon-btn weui-icon-btn_close"
          >
            关闭
          </wx-view>
        </wx-view>
        <wx-view
          class="weui-half-screen-dialog__hd__main"
        >
          <wx-view
            class="weui-half-screen-dialog__title"
          >
            <wx-view
              slot="title"
            >
              测试标题A
            </wx-view>
          </wx-view>
        </wx-view>
        <wx-view
          class="weui-half-screen-dialog__hd__side"
        >
          <wx-view
            class="weui-icon-btn weui-icon-btn_more"
          >
            更多
          </wx-view>
        </wx-view>
      </wx-view>
      <wx-view
        class="weui-half-screen-dialog__bd"
      >
        <wx-view
          slot="desc"
        >
          这是自定义描述区
        </wx-view>
      </wx-view>
      <wx-view
        class="weui-half-screen-dialog__ft"
      >
        <wx-view
          slot="footer"
        >
          <wx-button
            class="weui-btn"
            type="default"
          >
            按钮A
          </wx-button>
          <wx-button
            class="weui-btn"
            type="primary"
          >
            按钮B
          </wx-button>
        </wx-view>
      </wx-view>
    </wx-view>
  </wx-view>
</main>
`;

exports[`grids basic 2`] = `
<main>
  <wx-view
    class="weui-show"
  >
    <wx-view
      class="weui-mask init"
      data-type="tap"
      bind:tap="close"
    />
    <wx-view
      class="weui-half-screen-dialog "
    >
      <wx-view
        class="weui-half-screen-dialog__hd"
      >
        <wx-view
          class="weui-half-screen-dialog__hd__side"
          data-type="close"
          bind:tap="close"
        >
          <wx-view
            class="weui-icon-btn weui-icon-btn_close"
          >
            关闭
          </wx-view>
        </wx-view>
        <wx-view
          class="weui-half-screen-dialog__hd__main"
        >
          <wx-view
            class="weui-half-screen-dialog__title"
          >
            <wx-view
              slot="title"
            >
              测试标题A
            </wx-view>
          </wx-view>
        </wx-view>
        <wx-view
          class="weui-half-screen-dialog__hd__side"
        >
          <wx-view
            class="weui-icon-btn weui-icon-btn_more"
          >
            更多
          </wx-view>
        </wx-view>
      </wx-view>
      <wx-view
        class="weui-half-screen-dialog__bd"
      >
        <wx-view
          slot="desc"
        >
          这是自定义描述区
        </wx-view>
      </wx-view>
      <wx-view
        class="weui-half-screen-dialog__ft"
      >
        <wx-view
          slot="footer"
        >
          <wx-button
            class="weui-btn"
            type="default"
          >
            按钮A
          </wx-button>
          <wx-button
            class="weui-btn"
            type="primary"
          >
            按钮B
          </wx-button>
        </wx-view>
      </wx-view>
    </wx-view>
  </wx-view>
</main>
`;
